geoips/plugin_registry.py

```diff
@@ -0,0 +1,303 @@
+                                    plugin_type,
+                                    interface,
+                                    (plugin, subplg),
+                                    current_registry[plugin_type][interface][
+                                        plugin
+                                    ][subplg],
+                                )
+                        else:
+                            self.validate_plugin_attrs(
+                                plugin_type,
+                                interface,
+                                plugin,
+                                current_registry[plugin_type][interface][plugin],
+                            )
+                    except PluginRegistryError as e:
+                        print(e)
+                        pytest.xfail(str(e))
+
+    def validate_plugin_attrs(self, plugin_type, interface, name, plugin):
+        """Test non-product plugin for all required attributes."""
+        missing = []
+        if plugin_type == "yaml_based" and interface != "products":
+            attrs = [
+                "docstring",
+                "family",
+                "interface",
+                "package",
+                "plugin_type",
+                "relpath",
+            ]
+        elif plugin_type == "yaml_based":
+            attrs = [
+                "docstring",
+                "family",
+                "interface",
+                "package",
+                "plugin_type",
+                "product_defaults",
+                "source_names",
+                "relpath",
+            ]
+        else:
+            attrs = [
+                "docstring",
+                "family",
+                "interface",
+                "package",
+                "plugin_type",
+                "signature",
+                "relpath",
+            ]
+        for attr in attrs:
+            try:
+                plugin[attr]
+            except KeyError:
+                missing.append(attr)
+        if missing:
+            raise PluginRegistryError(
+                f"Plugin '{name}' is missing the following required "
+                f"top-level properties: '{missing}'"
+            )
+
+    def validate_registry_interfaces(self, current_registry):
+        """Test Plugin Registry interfaces validity."""
+        yaml_interfaces = [
+            "feature_annotators",
+            "gridline_annotators",
+            "product_defaults",
+            "products",
+            "sectors",
+        ]
+        module_interfaces = [
+            "algorithms",
+            "colormappers",
+            "coverage_checkers",
+            "filename_formatters",
+            "interpolators",
+            "output_checkers",
+            "output_formatters",
+            "procflows",
+            "readers",
+            "sector_adjusters",
+            "sector_metadata_generators",
+            "sector_spec_generators",
+            "title_formatters",
+        ]
```

Comment:
Can we avoid explicitly listing these? @evrose54

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/392/files#r1379308126
Address that separately if required. Closing this conversation.